src/Request.hs

```diff
@@ -126,9 +128,11 @@ mkValidator contractInfo@ContractInfo{..} dat lenderTn ctx = validate
 
     isItToCollateral :: TxOut -> Bool
     isItToCollateral txo = case toValidatorHash $ txOutAddress txo of
-      Just vh -> vh == collateralcsvh
+      Just vh -> case toValidatorHash collateralSc of
```

Comment:
Similarly to the other comment, this check extracts just the validator hash from the address and checks the same logic as before. Instead, I suggest to compare the whole address (`collateralSc` to the `txOutAddress txo`) which includes the staking part as well. You could also extract this into a handy util function

Reply:
https://github.com/aadafinance/aada-finance/pull/32/commits/e14cb159a3d6cad1b49d5b93aa19f285ab7d4b6c